src/gobupload/storage/handler.py

```diff
@@ -471,7 +471,10 @@ def has_any_entity(self, key=None, value=None):
         return self.session.query(self.DbEntity).filter_by(**{key: value}).count() > 0
 
     def get_collection_model(self):
-        return self.gob_model.get_collection(self.metadata.catalogue, self.metadata.entity)
+        try:
+            return gob_model[self.metadata.catalogue]['collections'][self.metadata.entity]
+        except KeyError:
+            return None
```

Comment:
Same repeated pattern again. Does not make it easier to read if you ask me.

Reply:
Added functions to verify messages and extract valid catalogs or collections.